### PR TITLE
Fix total paid by customer when he got partial refund

### LINKED ISSUE #15522: Partial refund - total paid by customer is not correct
**Describe the bug**
When we create a new customer and a new order for this customer (with total for exemple 100 euro), as the order's status is "waiting for payement" the total paid will be 0 (correct). Change the order's status to "payement accepted", the total paid wil be change to 100 euro (correct). Then, try to refund one of the products (50 euro), and change the status to "refunded". Usually, the total will be 0 and that's wrong cause the amount of refund is 50 euro and not 100 euro

**To Reproduce**

1. Create a new customer
1. Create a new order for this customer with total = 100 euro
1. Go to the BO > Order details page => change status to payement accepted
1. Click on partial refund to refund one of the products (50 euro)
1. Change the status of the order to refunded
1. check the `Total spent since registration` = 0 => NOK => the amount of refund is 50 euro and not 100 euro => it should be 50

**Screenshots**
https://drive.google.com/file/d/1fdXB-a3i8v_zPBp0DtRhfidSKobu8Niq/view

**Additional information**
PrestaShop version: 1.7.6.1
PHP version: 7.2

### PR DESCRIPTION
**Pull request by hibatallahAouadni**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.6.1.x
| Description?  | I added two SQL queries to combine the resulted arrays "resultOrders" and "resultOrdersSlips" and to get the correct total paid by customer.
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | http://forge.prestashop.com/browse/PSCSX-8285 & Fixes #15522
| How to test?  | create a new customer and a new order for this customer (with total for exemple 100 euro), as the order's status is "waiting for payement" the total paid will be 0 (correct). Change the order's status to "payement accepted", the total paid wil be change to 100 euro (correct). Then, try to refund one of the products (50 euro), and change the status to "refunded". Usually, the total will be 0 and that's wrong cause the amount of refund is 50 euro and not 100 euro, with the new changement you will see that the total is 50 euro (correct). {see the attachement bellow}

refunded

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/8239)

### CODE REVIEW

File: classes/Customer.php [hunk 1]
```diff
@@ -550,22 +550,43 @@ public static function searchByIp($ip)
...
+        GROUP BY o.id_order
+          ');
+        $resultOrderSlip = Db::getInstance()->executeS('
+        SELECT o.`id_order`, MAX(o.total_paid) AS total_paid, SUM(s.total_products_tax_incl) AS total_product_tax_incl ,SUM(s.total_shipping_tax_incl) AS total_shipping_tax_incl , MAX(o.`conversion_rate`) AS conversion_rate
+        FROM `' . _DB_PREFIX_ . 'orders` o
+        INNER JOIN `' . _DB_PREFIX_ . 'order_slip` s ON o.id_order = s.id_order
+        WHERE o.`id_customer` = ' . (int)$this->id . '
+        AND o.valid = 0
+          GROUP BY o.id_order
+        ');
```
Comment (hibatallahAouadni) [inline classes/Customer.php:569]: @Quetzacoalt91 code review please :relaxed: 
Thanks in advance.

Comment (hibatallahAouadni) [inline classes/Customer.php]: @Quetzacoalt91 code review please :relaxed: 
Thanks in advance.

**Comment by hibatallahAouadni** [comment]
@vincentbz @eternoendless milestone 1.6.1.18 please

**Comment by matks** [comment]
@marionf if this bug is still happening in 1.7, we might be able to fix it thanks to this PR but we will need extensive Behat tests to make sure we do not introduce a regression in some edge-case for customer stats

**Comment by marionf** [comment]
@khouloudbelguith Could you try to reproduce it with 1.7.6.1 ?

**Comment by khouloudbelguith** [comment]
Hi @marionf,

I manage to reproduce the issue with PS1.7.6.1.
https://drive.google.com/file/d/1fdXB-a3i8v_zPBp0DtRhfidSKobu8Niq/view
Issue created here: https://github.com/PrestaShop/PrestaShop/issues/15522

Thanks!
